Flagsmith.FlagsmithClient/Flags.cs

```diff
@@ -34,17 +36,19 @@ public async Task<Flag> GetFlag(string featureName)
                 await _AnalyticsProcessor.TrackFeature(flag.GetFeatureName());
             return flag;
         }
-        public List<Flag> AllFlags() => _Flags;
+
+        public Task<IReadOnlyCollection<IFlag>> AllFlags() => Task.FromResult<IReadOnlyCollection<IFlag>>(_Flags);
+
         private static Flag FromFeatureStateModel(FeatureStateModel featureStateModel, string identityId = null) =>
-            new Flag(new Feature(featureStateModel.Feature.Name, featureStateModel.Feature.Id), featureStateModel.Enabled, featureStateModel.GetValue(identityId)?.ToString(), featureStateModel.Feature.Id);
+            new Flag(new Feature(featureStateModel.Feature.Name, featureStateModel.Feature.Id), featureStateModel.Enabled, featureStateModel.GetValue(identityId)?.ToString());
 
-        public static Flags FromFeatureStateModel(AnalyticsProcessor analyticsProcessor, Func<string, Flag> defaultFlagHandler, List<FeatureStateModel> featureStateModels, string identityId = null)
+        public static Flags FromFeatureStateModel(AnalyticsProcessor analyticsProcessor, Func<string, IFlag> defaultFlagHandler, List<FeatureStateModel> featureStateModels, string identityId = null)
```

Comment:
On the same logic as the above suggestion, this should be `FromFeatureStateModels`.

Reply:
same here